### ISSUE TITLE
fails to write flat POM if outputDirectory doesn't exist

### ISSUE THREAD

**Issue by thendarion** [outbrain/ci-friendly-flatten-maven-plugin#11]
# Expected behaviour
I can specify any output directory for the flat POM and it is created if it doesn't exist.

# Actual behaviour
I can only specify existing directories, otherwise the plugin fails.

**Comment by ayoukhananov** [comment]
Hi @thendarion 
I am not sure you are able to pass the outputDirectory in the current implementation,
Since it has default path points to "${project.basedir}".
I didn't understand what are you trying to do ? I will appreciate if you could explain 

Waiting forward to your response

**Comment by thendarion** [comment]
for example if I set 
`<outputDirectory>target</outputDirectory>`
and I run
`mvn clean ci-friendly-flatten:flatten`
then I get
```
--- ci-friendly-flatten-maven-plugin:1.0.8:flatten (default-cli) @ project ---
[INFO] Replacing CI friendly properties for project ...
[ERROR] An error occurred while writing C:\project\target\.ci-friendly-pom.xml
java.io.FileNotFoundException: C:\project\target\.ci-friendly-pom.xml (The system cannot find the path specified)
    at java.io.FileOutputStream.open0 (Native Method)
    at java.io.FileOutputStream.open (FileOutputStream.java:292)
    at java.io.FileOutputStream.<init> (FileOutputStream.java:235)
    at java.io.FileOutputStream.<init> (FileOutputStream.java:185)
    at java.io.FileWriter.<init> (FileWriter.java:96)
    at com.outbrain.ci.friendly.flatten.maven.plugin.FlattenMojo.writePom (FlattenMojo.java:63)
    at com.outbrain.ci.friendly.flatten.maven.plugin.FlattenMojo.execute (FlattenMojo.java:56)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:567)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
```

so outputdirectory is read and used but when you write the file, you fail because the folder doesn't exist :(

**Comment by ayoukhananov** [comment]
@thendarion Issue fixed and released in 1.0.9 version

**Comment by thendarion** [comment]
ty :)
